pkg/dependencies/factory.go

```diff
@@ -770,6 +771,24 @@ func (f *Factory) WithHelm(opts ...executables.HelmOpt) *Factory {
 	return f
 }
 
+func (f *Factory) WithHelmFactory(opts ...executables.HelmOpt) *Factory {
+	f.WithExecutableBuilder()
+
+	f.buildSteps = append(f.buildSteps, func(ctx context.Context) error {
```

Comment:
Overall, factory is not really a preferred abstract in golang. As we have adopted it, we should try not making it more complicated.

This method introduces something feels like "nested factory", which makes the abstract harder to understand. I believe we can achieve the target without this "helmFactory"